include/spatio_temporal_voxel_layer/spatio_temporal_voxel_layer.hpp

```diff
@@ -126,6 +130,11 @@ class SpatioTemporalVoxelLayer : public costmap_2d::CostmapLayer
   bool AddStaticObservations(const observation::MeasurementReading& obs);
   bool RemoveStaticObservations(void);
 
+  // Dynamic reconfigure
+  void DynamicReconfigureCallback(dynamicReconfigureType &config, uint32_t level);
+  dynamicReconfigureServerType* _dynamic_reconfigure_server;
+  bool _first_time;
```

Comment:
I don't understand the need for this variable

Reply:
When dynamic reconfigure server gets created, it calls the callback with the default configuration (with de values specified in the .cfg) so is common to handle this specific case separately. In other projects I use this as a initialization-like method too, but for this case I think is better to use the rosparams as (1) follows what you might have already and (2) it follows the move\_base way of configuring the system.

This variable could be removed if the server is initialized before all the variables that are got nowadays with rosparam but it might create race conditions. E.g. the code has a variable called "start\_working" this will get initialized 3 times: 
1. in the constructor of the object
2. in the dynamic reconfigure server
3. in rosparam

If this variable is used in another thread to do something this might create problems depending on where the thread is created. I put this variable in case it wanted to be used for initializing something, but it can be removed if the server is initialized before the rosparams.